### ISSUE TITLE
Error packaging latest release for NixOS; possible bug in `setup.py`

### ISSUE THREAD

**Issue by paperdigits** [eoyilmaz/displaycal-py3#150]
I'm trying to produce a package for NixOS, a declarative linux distribution. To make packages more consistent, the build tool sets the build time and date to all zeros. But when I try and build displaycal, I get the following error:

```
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/build/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 1587, in <module>
    setup()
  File "setup.py", line 563, in setup
    lastmod_time = calendar.timegm(time.strptime(lastmod, "%Y-%m-%dT%H:%M:%SZ"))
  File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 562, in _strptime_time
    tt = _strptime(data_string, format)[0]
  File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '0000-00-00T00:00:00.0Z' does not match format '%Y-%m-%dT%H:%M:%SZ'
error: builder for '/nix/store/arz8971jv0iqazlf0ax8wvh23z3wrxi9-python3.10-displaycal-py3-3.9.6.drv' failed with exit code 1;
       last 10 log lines:
       >     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
       >   File "setup.py", line 1587, in <module>
       >     setup()
       >   File "setup.py", line 563, in setup
       >     lastmod_time = calendar.timegm(time.strptime(lastmod, "%Y-%m-%dT%H:%M:%SZ"))
       >   File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 562, in _strptime_time
       >     tt = _strptime(data_string, format)[0]
       >   File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 349, in _strptime
       >     raise ValueError("time data %r does not match format %r" %
       > ValueError: time data '0000-00-00T00:00:00.0Z' does not match format '%Y-%m-%dT%H:%M:%SZ'
       For full logs, run 'nix log /nix/store/arz8971jv0iqazlf0ax8wvh23z3wrxi9-python3.10-displaycal-py3-3.9.6.drv'.
```

So `lastmod` is being set using https://github.com/arkordt/displaycal-py3/blob/d42be4b223d1e3e469063a0592223afd35593486/setup.py#L452

Specifically the problem is line 487 where they append `str(round(mtime - int(mtime), 6))[1:]` to the lastmod string

When it's reparsed on line 559 here: https://github.com/arkordt/displaycal-py3/blob/d42be4b223d1e3e469063a0592223afd35593486/setup.py#L559

There's a decimal and some digits that aren't expected by this format string.

Any direction on how to fix this would be very helpful!

**Comment by p5k369** [comment]
Hey @paperdigits, the parsing of the string has been updated at some point but not the generation so yes this is a bug, however I think you encounter the problem because of the default that is set in `meta.py`.
Take a look at #152 and see if that fixes your problem.

**Comment by paperdigits** [comment]
Hi @p5k369 thank you so much for taking the time.

I still get a similar error, though the formatting of the time stamp looks good now:

```
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/build/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 1583, in <module>
    setup()
  File "setup.py", line 559, in setup
    lastmod_time = calendar.timegm(time.strptime(lastmod, "%Y-%m-%dT%H:%M:%SZ"))
  File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 562, in _strptime_time
    tt = _strptime(data_string, format)[0]
  File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '0000-00-00T00:00:00Z' does not match format '%Y-%m-%dT%H:%M:%SZ'
error: builder for '/nix/store/f6gy21qphkkx2m6bkdc8kkfx24g5m32q-python3.10-displaycal-py3-3.9.6.drv' failed with exit code 1;
       last 10 log lines:
       >     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
       >   File "setup.py", line 1583, in <module>
       >     setup()
       >   File "setup.py", line 559, in setup
       >     lastmod_time = calendar.timegm(time.strptime(lastmod, "%Y-%m-%dT%H:%M:%SZ"))
       >   File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 562, in _strptime_time
       >     tt = _strptime(data_string, format)[0]
       >   File "/nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/lib/python3.10/_strptime.py", line 349, in _strptime
       >     raise ValueError("time data %r does not match format %r" %
       > ValueError: time data '0000-00-00T00:00:00Z' does not match format '%Y-%m-%dT%H:%M:%SZ'
       For full logs, run 'nix log /nix/store/f6gy21qphkkx2m6bkdc8kkfx24g5m32q-python3.10-displaycal-py3-3.9.6.drv'.
```

**Comment by p5k369** [comment]
I think the problem is that the fallback date is invalid. To be honest I do not know why it was chosen. But setting it to a valid date should do the trick, for example `1970-01-01T00:00:00Z`.

**Comment by paperdigits** [comment]
@p5k369 yes you are right! Thank you so much for patching this issue.

**Comment by eoyilmaz** [comment]
I guess this is fixed with @p5k369 's MR of #152
Thank you @p5k369 :+1:
